### ISSUE TITLE
False positive for repeated-keywords when dict item is popped off 

### ISSUE THREAD

**Issue by amrishparmar** [PyCQA/pylint#7614]
### Bug description

We found that if a dictionary is unpacked as arguments to a function, pylint still flags the `repeated-keywords` even if a pop happens on the dictionary in between (meaning that no kwargs are actually repeated)

```python
def foo(a=None, b=None, c=None):
    pass

d = {"a": 1, "b": 2, "c": 3}

x = d.pop("c")  

foo(c=x, **d)

```

### Command used

```shell
pylint --disable=all --enable=repeated-keyword test.py
```

### Pylint output

```shell
************* Module test
test.py:8:0: E1132: Got multiple values for keyword argument 'c' in function call (repeated-keyword)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No error is flagged for the line

### Pylint version

```shell
pylint 2.15.4
astroid 2.12.11
Python 3.10.6 (main, Oct  7 2022, 20:19:58) [GCC 11.2.0]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. It's probably blocked by #4795

**Comment by atkretsch** [comment]
I know this is an old issue, but I'd like to point out that this same basic issue can result in false *negatives* as well.

Example:
```python
# test.py

def foo(a=None, b=None, c=None):
    pass

d = {"a": 1, "b": 2}

d["c"] = 3

foo(c=4, **d)
```

Command used:
```
pylint --disable=all --enable=repeated-keyword test.py
```

Pylint output:
```
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 4.00/10, +6.00)
```

Expected behavior:

Should get output something like this:

```
************* Module test
test.py:10:0: E1132: Got multiple values for keyword argument 'c' in function call (repeated-keyword)
```

Pylint version:
```
pylint 3.2.7
astroid 3.2.4
Python 3.9.18 (main, May 30 2024, 09:59:23) 
[Clang 15.0.0 (clang-1500.3.9.4)]
```

**Comment by e-gebes** [comment]
I ran into the same issue, false positive with a dict pop.

If #4795 is out of reach for now, is there a possibility to add certain heuristic for the most common cases where on the exact name of the object `<name>.pop["c"]` / `del <name>["c"]` (false positive) or `d.update(...)` / `d["c"] = ...` (false negative) is applied? If that makes sense or is worth the effort is up to you, I just wanted to mention the idea.

Thank you for your efforts, Pylint is a useful tool!
